terraform/modules/cloud-build-docker/cloudbuild.yml

```
@@ -0,0 +1,36 @@
+# Cloud Build configuration with branch-based caching.
+# Uses branch name as cache tag for optimal layer reuse.
+steps:
+  - name: 'gcr.io/cloud-builders/docker'
+    id: Pull previous image
+    entrypoint: bash
+    args: ['-c', 'docker pull $_IMAGE_NAME:$_CACHE_TAG || true']
+
+  - name: 'gcr.io/cloud-builders/docker'
+    id: Build image
+    entrypoint: bash
+    args:
+      - -c
+      - |
+        if docker image inspect $_IMAGE_NAME:$_CACHE_TAG > /dev/null 2>&1; then
+          echo "Using cache from $_IMAGE_NAME:$_CACHE_TAG"
+          docker build --tag=$_IMAGE_TAG --cache-from=$_IMAGE_NAME:$_CACHE_TAG --build-arg BASE_IMAGE=$_BASE_DIGEST .
```

This is just a shell command right?  So do we need to do quoting?
```suggestion
          docker build --tag="$_IMAGE_TAG" --cache-from="$_IMAGE_NAME:$_CACHE_TAG" --build-arg BASE_IMAGE="$_BASE_DIGEST" .
```
and likewise with the other shell commands in this file.

---

These are Cloud Build substitutions that get replaced with actual values before the shell command runs. The shell handles these as regular environment variables, so no special quoting is needed.

---

But the cloud-build substitution could still have space.  What if `$_IMAGE_NAME` was `my name here`?  Wouldn't we need to quote it to make sure the shell interpreted it properly?

---

Fair, I'll give the quotes a try.